client/http_sender.go

```diff
@@ -125,14 +125,11 @@ func (s *httpSender) Send(_ context.Context, call Call) {
 				// the errors we'll sweep up in this net shouldn't be retried,
 				// but we can't really know for sure which.
 				log.Warningf("failed to send HTTP request or read its response: %s", t)
-				return retry.Continue, nil
-			default:
-				// Can't retry in order to recover from this error. Propagate.
-				return retry.Break, err
+				return retry.Continue, err
 			}
 		}
-		// On successful post, we're done with retry loop.
-		return retry.Break, nil
+		// If not previously handled, propagate the error (which may be nil).
```

Comment:
I disagree; it's an important part of go's error-handling idioms that you check `error` values and in general spend as little time as possible with an indeterminate `err`. An `if err != nil` block that does not return (or otherwise completely handle the error) invites bugs if something comes along and assumes that it is now in a "success" code path. `return retry.Break, nil` doesn't need any more scrutiny in a retryable function than `return nil` does in a regular function.